Public/Src/Cache/ContentStore/Distributed/ContentLocations/IContentLocationStore.cs

```diff
@@ -114,6 +114,11 @@ public interface IContentLocationStore : IStartupShutdown
         /// Reports about new reputation for a given location.
         /// </summary>
         void ReportReputation(MachineLocation location, MachineReputation reputation);
+
+        /// <summary>
+        /// Returns a random machine location, excluding the specified location. Returns default if operation is not possible.
+        /// </summary>
+        MachineLocation GetRandomMachineLocation(MachineLocation except);

```

Comment:
Do you want to name it 'currentMachine'? Because I suspect we just want to exclude the current machine.

If we'll use 'currentMachine' then the method will be slightly less generic, but imo it would be clearer.

Reply:
I think "except" better conveys the behavior or the function, whereas "currentMachine" doesn't tell me anything about what the implementation will do with it

Reply:
I think "except" better conveys the behavior or the function, whereas "currentMachine" doesn't tell me anything about what the implementation will do with it

---
In reply to: [312215220](https://github.com/microsoft/BuildXL/pull/705#discussion_r312215220) [](ancestors = 312215220)